cdap-app-templates/cdap-etl/hydrator-test/src/main/java/io/cdap/cdap/etl/mock/action/FieldLineageAction.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.mock.action;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import com.google.gson.reflect.TypeToken;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.lineage.field.Operation;
+import io.cdap.cdap.api.lineage.field.OperationType;
+import io.cdap.cdap.api.lineage.field.ReadOperation;
+import io.cdap.cdap.api.lineage.field.TransformOperation;
+import io.cdap.cdap.api.lineage.field.WriteOperation;
+import io.cdap.cdap.api.plugin.PluginClass;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.api.plugin.PluginPropertyField;
+import io.cdap.cdap.data2.metadata.lineage.AccessType;
+import io.cdap.cdap.etl.api.action.Action;
+import io.cdap.cdap.etl.api.action.ActionContext;
+import io.cdap.cdap.etl.proto.v2.ETLPlugin;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Mock action to write dummy field lineage information for two datasets
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("FieldLineageAction")
+public class FieldLineageAction extends Action {
+  public static final PluginClass PLUGIN_CLASS = getPluginClass();
+
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(Operation.class, new OperationTypeAdapter()).create();
+  private static final Type LIST_OPERTAION = new TypeToken<List<Operation>>() { }.getType();
+  private final Config config;
+
+  /**
+   * Config for the FieldLineageAction
+   */
+  public static class Config extends PluginConfig {
+    private String readDataset;
+    private String writeDataset;
+    // json string for the field operations
+    private String fieldOperations;
+  }
+
+  public FieldLineageAction(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    context.registerLineage(config.readDataset, AccessType.READ.name());
+    context.registerLineage(config.writeDataset, AccessType.WRITE.name());
+
+    List<Operation> operations = GSON.fromJson(config.fieldOperations, LIST_OPERTAION);
+    context.record(operations);
+  }
+
+  public static ETLPlugin getPlugin(String readDataset, String writeDataset, String fieldOperations) {
```

Comment:
it's better for this to take a `Collection<Operation>` instead of a string, so that all the logic for serialization/deserialization is encapsulated in this class instead of spread out across the caller and this class.